1-JavaScript/Week2/homeworkSolition/ex6-sol-totalCost.js

```diff
@@ -0,0 +1,44 @@
+'use strict';
+const cartForParty = {
+  beers: 1.75,
+  chips: 0.99,
+  chikenwings: 2.433,
+  water: 2,
+  bread: 1.5,
+};
+
+function calculateTotalPrice(price) {
+  // TODO replace this comment with your code
+  // let sum = 0
+  // for(let x of price){
+  //   sum = sum + x
+  // }
+  const initial = 0
+  const sum = price.reduce((acc,val)=> acc+val+initial)
```

Comment:
Awesome, reducer 💪 

So with `reduce` you don't need to add the initial value as well 🙂 In this case it will do no harm because initial is 0, but what if it's 10? Or 20? Then every time you add that initial value to the value you're calculating. A reducer function actually accepts an initial value already. If you leave it empty, the function is initialised with the first value in the array and then start counting from the second. But I like to specify it explicitly 🙂 

```suggestion
  const sum = price.reduce((acc, val)=> acc + val, 0)
```